### PR TITLE
Eliminate DomainUsage setting

### LINKED ISSUE #657: Eliminate DomainUsage option
When running tests under .NET Framework, the user may specify running multiple assemblies in one domain or in a separate domain per assembly. This complicates coordinating the test results when multiple domains are used and serves no particular purpose from the user point of view. (Note that we don't run multiple domains in parallel, only multiple processes)

While we can't eliminate use of AppDomains (some platforms require it), we can eliminate the user's ability to decide how AppDomains are used. This feature would eliminate the menu item, which allows selecting and would result in subsequently eliminating quite a lot of code that would no longer be used.

### PR DESCRIPTION
**Pull request by CharliePoole**
Fixes #657

### CODE REVIEW

**Comment by CharliePoole** [comment]
@mikkelbu @immeraufdemhund This is a pretty big change so I'd appreciate it if one or both of you can review it. Since the tests all pass, the main question is whether there's anything more subtle from a user perspective that I've missed.

BTW, I gave some thought to the scope of the change. I could have done it in the gui runner alone, leaving the capability to set the DomainUsage still present in the engine. I decided against it. I think it's really up to the engine to figure out internally the best way to isolate tests, so this is something that should be outside the control of both the user and the test runner.

**Comment by immeraufdemhund** [comment]
Today is my first day back for extra activities. I've been training a new intern on React and wiring keyboards. I'll take a look at this sometime today

**Comment by mikkelbu** [comment]
@CharliePoole I should have time to review it this weekend.

**Review by mikkelbu — COMMENTED** [review]
The changes look sound and I only found a couple of places where I think we could clean up some more test cases. My only question is whether this change will confuse a user that specifies `DomainUsage` via an *.nunit file and this is then overriden by the default. It has been a long time since I last time looked closely at the code, but can (and should) we display a notification/warning/... when this happens?

**Review by mikkelbu — APPROVED** [review]

File: src/TestModel/tests/TestModelCreationTests.cs [hunk 1]
```diff
@@ -45,7 +45,6 @@ public void CreateTestModel(params string[] args)
 
             var checker = new PackageOverridesChecker(model);
             checker.CheckSetting(options.ProcessModel, EnginePackageSettings.ProcessModel);
-            checker.CheckSetting(options.DomainUsage, EnginePackageSettings.DomainUsage);
```
[the remark this answers is not in the document]
Comment (CharliePoole) [reply src/TestModel/tests/TestModelCreationTests.cs:48]: Actually, I found a bunch I missed in the Experimental GUI. I should  remove it there as well. Once the CI passes, I'll merge.

**Comment by CharliePoole** [comment]
@mikkelbu Good point about NUnit project files. I'll create a new issue for that since this one is big enough already.

File: src/TestModel/tests/TestModelCreationTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikkelbu) [reply src/TestModel/tests/TestModelCreationTests.cs:48]: I was going to say that I didn't find any. The only project with DomainUsage was the ProjectEditor :)

**Pull request merged**